elisp.nix

```diff
@@ -40,6 +40,8 @@ let
       else if type == "string" then config
       # - A config path { config = ./config.el; }
       else if type == "path" then builtins.readFile config
+      # - A list of paths { config = [ ./config.el ./custom.el ] }
+      else if type == "list" then lib.concatMapStringsSep "\n" (fname: builtins.readFile fname) config
```

Comment:
Eta reduction
```suggestion
      else if type == "list" then lib.concatMapStringsSep "\n" builtins.readFile config
```

Reply:
re https://github.com/nix-community/emacs-overlay/pull/465#issuecomment-2631268955

Presumably because you do not apply my suggestion as is and instead an extra `map` is added which breaks the code.

Reply:
I did, bot `lib.concatMapStringsSep` and the indirect `lib.concatStringsSep (map ...` result in unresolved `thunk`s. This commit only uses the latter because in a narrow scenario that *did* resolve them, whereas `lib.concatMapStringsSep` does not. I'd figured that it's maybe because `lib.concatMapStringsSep` demands specifically strings from its input list, but that's clearly not it.